src/agent/src/policy.rs

```diff
@@ -243,6 +249,70 @@ impl AgentPolicy {
             }
         }
     }
+
+    pub async fn get_data(&mut self, path: &str) -> Result<String> {
+        debug!(sl!(), "policy read: {path}");
+        self.log_opa_input(path, EMPTY_JSON_INPUT).await;
+
+        if let Some(opa_client) = &mut self.opa_client {
+            let uri = format!("{}{path}", &self.query_path);
+            let response = opa_client
+                .post(uri.clone())
+                .body(EMPTY_JSON_INPUT.to_string())
+                .send()
+                .await?;
+
+            if response.status() != http::StatusCode::OK {
+                bail!(
+                    "policy read: POST {uri} response status {}",
+                    response.status()
+                );
+            }
+
+            let http_response = response.text().await?;
+            let trimmed_response = http_response.trim();
+
+            if trimmed_response == "{}" {
+                let error = format!("policy read: <{uri}> not found in policy");
+                debug!(sl!(), "{error}");
+                bail!(error)
+            } else {
+                let opa_response: serde_json::Result<StringResponse> =
+                    serde_json::from_str(&http_response);
+                match opa_response {
+                    Ok(resp) => {
+                        debug!(sl!(), "policy read: {path} -> <{}>", &resp.result);
+                        Ok(resp.result)
+                    }
+                    Err(e) => bail!("policy read: POST {uri} serde error {e}"),
+                }
+            }
+        } else {
+            bail!("Agent Policy is not initialized")
+        }
+    }
+
+    async fn get_string_value(&mut self, key: &str) -> Result<String> {
+        Ok(self.get_data(key).await?)
+    }
+
+    // TODO: remove this test
+    pub async fn test_query_policy_data(&mut self) {
+        // See kata-containers/tests/integration/kubernetes/runtimeclass_workloads/k8s-policy-set-keys.rego
+        // for an example of how to add this data into your policy.
+        let key = "my_test_data/default/key/ssh-demo";
```

Comment:
This looks like a key, which I don't think we should be passing via policy. Might be better to do something more like a public key here.

Reply:
It's possible that a better example would be more helpful. But the only goal for this code is to show how to read arbitrary data from Policy (the my_test_data/default/key/ssh-demo data in this case).

Here I just borrowed some example data from https://github.com/confidential-containers/confidential-containers/blob/b36a57e530aa83741fba2592574a7a4a4d8892b6/demos/ssh-demo/aa-offline_fs_kbc-keys.json or similar JSON file.

Reply:
Yeah no big deal, but that is a symmetric key. Might help make your example clearer to use something else.